mixer/adapter/list/regexList.go

```diff
@@ -23,9 +24,14 @@ type regexList struct {
 	regexpList []*regexp.Regexp
 }
 
-func (l *regexList) checkList(symbol string) (bool, error) {
+func (l *regexList) checkList(symbol interface{}) (bool, error) {
+	if _, ok := symbol.(string); !ok {
+		// invalid symbol format
+		return false, fmt.Errorf("%v is not a valid string", symbol)
+	}
+
 	for _, exp := range l.regexpList {
-		if exp.MatchString(symbol) {
+		if exp.MatchString(symbol.(string)) {
```

Comment:
I don't know if the compiler takes care of this, but it's probably better to put the type cast outside of the for loop, since type casting involves a few lookups.